From 902828a6af170305c3ab5148f45c804cb1b5132a Mon Sep 17 00:00:00 2001
From: Yuxi <b36102@freescale.com>
Date: Tue, 25 Oct 2011 16:25:30 +0800
Subject: [PATCH 235/449] ENGR00160728 fix deranged image on mx6q platform
 when doing overlay

Set the right output frame buffer for mxc_v4l2_overlay unit test

Signed-off-by: Yuxi Sun <b36102@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_overlay.c | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_overlay.c b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
index e2ed6ac..af4b690 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_overlay.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
@@ -448,12 +448,6 @@ mxc_v4l_overlay_setup(struct v4l2_format *fmt)
 	printf("frame_rate is %d\n",
 	       parm.parm.capture.timeperframe.denominator);
 
-        if (ioctl(fd_v4l, VIDIOC_S_OUTPUT, &g_display_lcd) < 0)
-        {
-                printf("VIDIOC_S_OUTPUT failed\n");
-                return TFAIL;
-        }
-
 	ctl.id = V4L2_CID_PRIVATE_BASE + 2;
 		ctl.value = g_rotate;
         if (ioctl(fd_v4l, VIDIOC_S_CTRL, &ctl) < 0)
@@ -645,6 +639,18 @@ main(int argc, char **argv)
                 return TFAIL;
         }
 
+	if (strcmp(fb0_fix.id, "DISP3 BG - DI1") == 0)
+		g_display_lcd = 1;
+	else if (strcmp(fb0_fix.id, "DISP3 BG") == 0)
+		g_display_lcd = 0;
+
+        if (ioctl(fd_v4l, VIDIOC_S_OUTPUT, &g_display_lcd) < 0)
+        {
+                printf("VIDIOC_S_OUTPUT failed\n");
+                return TFAIL;
+        }
+
+
 	fb_device_fg = "/dev/fb1";
 	if ((g_fd_fb_fg = open(fb_device_fg, O_RDWR)) < 0) {
 		printf("Unable to open frame buffer 1\n");
-- 
1.8.0

